codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSServiceConfig.kt

```diff
@@ -174,7 +178,11 @@ class AWSServiceConfig(writer: SwiftWriter, val ctx: ProtocolGenerator.Generatio
                 }
 
                 ENDPOINT_RESOLVER -> {
-                    writer.write("self.${it.memberName} = ${it.memberName} ?? DefaultEndpointResolver()")
+                    writer.openBlock("if let endpointResolver = endpointResolver {", "} else {") {
+                        writer.write("self.endpointResolver = endpointResolver")
+                    }
+                    writer.indent().write("self.${it.memberName} = try \$L()", AWSServiceTypes.DefaultEndpointResolver)
+                    writer.dedent().write("}")
```

Comment:
No need to address it here, but using indent then dedent is an awkward, potentially error-prone way of writing an if-else statement... wonder if there should be a method on writer for a two-block statement